tests/yast2_gui/yast2_firewall.pm

```diff
@@ -79,19 +91,17 @@ sub verify_service_stopped {
     assert_screen 'yast2_firewall_start-up';
     assert_screen 'yast2_firewall_service_status_running';
     send_key $fw{service_stop};
-    assert_screen [qw(yast2_firewall_service_status_stopped generic-desktop)];
-    if (match_has_tag('generic-desktop')) {
-        record_soft_failure "bsc#1114677 - Dialog dissapear after switching service status";
-    }
+    detect_dialog_flickering 'yast2_firewall_service_status_stopped';
     assert_screen 'yast2_firewall_service_status_stopped';
     wait_screen_change { send_key $cmd{accept} };
     assert_screen 'generic-desktop';
 
     select_console 'root-console';
     if (script_run("firewall-cmd --state 2>&1 | grep 'not running'") != 0) {
         record_soft_failure "bsc#1114807 - service does not stop ";
-        return;
+        return 1;
     }
+    return;
```

Comment:
If you return explicit values, let's be consistent and define them explicitly too. Also, it's a bit counter intuitive to return value which is `true` in perl in case bug was detected.